### DISCUSSION TITLE
Submit queries through GraphQL or like-wise interface

### DISCUSSION THREAD

**Discussion by vmesel** [prest/prest#826] [API Server]
A problem I've been working on in my personal projects is the possibility of joining multiple tables and doing multiple filters without limitations imposed by a query file as we have on pREST.

I've been thinking about implementing a GraphQL endpoint that works with all default aggregations depending on the data type and the possibility of joining tables based on the fields given to the API. (Or maybe a custom JSON query type, the only thing we should be able to pass is a list of fields and filters.

The type of JSON I was thinking of is:

```
{
    query: [
        "table1.field1",
        "sum:table2.field2",
        "avg:table3.field3"
    ],
    filters: [
{
    "filter_type": "where",
    "field_to_be_filtered": "table2.field4",
    "match_type": "equal_or_greater",
    "values": [2.5]
}
]
}

```

**Comment by c-nv-s** [comment]
I was was trying to find the same solution to what you are trying to achieve and also saw this issue: https://github.com/prest/prest/issues/353

Like you mention it doesn't strictly have to be Graphql, but just an endpoint where you can pass the necessary json payload.

I something akin to this:
```
{
  "limit": 25,
  "filter": {
    "admin": false,
    "created_at": {
      "$gt": "2018-01-01T16:00:00.000Z",
      "$lt": "2018-04-01T16:00:00.000Z"
    }
    "$or": [
      { "address": "TLV" },
      { "address": "NYC" }
    ]
  }
  "sort": ["-created_at"]
}
```
taken from https://github.com/a8m/rql#medium-example
